src/segment-loader.js

```diff
@@ -1231,22 +1235,47 @@ export default class SegmentLoader extends videojs.EventTarget {
     // can be adjusted by the timestamp offset
     if (!segmentInfo.hasAppendedData_) {
       this.metadataQueue_.caption.push(
-        this.handleCaptions_.bind(this, simpleSegment, captions));
+        this.handleCaptions_.bind(this, simpleSegment, captionData));
       return;
     }
 
     const timestampOffset = this.sourceUpdater_.videoTimestampOffset() === null ?
       this.sourceUpdater_.audioTimestampOffset() :
       this.sourceUpdater_.videoTimestampOffset();
 
-    captions.forEach((caption) => {
-      createCaptionsTrackIfNotExists(
-        this.inbandTextTracks_, this.hls_.tech_, caption.stream);
-      addCaptionData({
-        captionArray: [caption],
-        inbandTextTracks: this.inbandTextTracks_,
-        timestampOffset
-      });
+    const tracks = {};
+
+    // get total start/end and captions for each track/stream
+    captionData.forEach((caption) => {
+      // caption.stream is actually a track name...
+      // set to the existing values in tracks or default values
+      tracks[caption.stream] = tracks[caption.stream] || {
+        startTime: Infinity,
+        captions: [],
+        endTime: 0
```

Comment:
If these are defaults, should the values for `startTime` and `endTime` here be swapped?

Reply:
these defaults are a clever way of getting lines 1260-1261 to be accurate

Reply:
added a comment to clarify this

Reply:
Gotcha, gotcha. It just seemed backwards. I get it now.